lib/repositories/filler-address-repository.ts

```diff
@@ -0,0 +1,120 @@
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+import Logger from 'bunyan';
+import { Entity, Table } from 'dynamodb-toolbox';
+
+import { DYNAMO_TABLE_NAME } from '../constants';
+
+export interface FillerAddressRepository {
+  getFillerAddresses(filler: string): Promise<string[] | undefined>;
+  getFillerByAddress(address: string): Promise<string | undefined>;
+  addNewAddressToFiller(address: string, filler?: string): Promise<void>;
+}
+/*
+ * Dynamo repository for managing filler addresses
+ * Supports two way lookups: filler -> addr, addr -> fillers
+ */
+export class DynamoFillerAddressRepository implements FillerAddressRepository {
+  static log: Logger;
+
+  static create(documentClient: DynamoDBDocumentClient): FillerAddressRepository {
+    this.log = Logger.createLogger({
+      name: 'FillerAddressRepository',
+      serializers: Logger.stdSerializers,
+    });
+
+    const addressTable = new Table({
+      name: DYNAMO_TABLE_NAME.FILLER_ADDRESS,
+      partitionKey: 'pk', // generic partition key name to support both filler and address
```

Comment:
It looks like DynamoDB supports secondary indexes so we don't need to store two separate records just for indexing:
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html

Reply:
Is the suggestion that we store one `filler<>address` relation per item, make one the partition key and the other the primary key of the index? I feel like that's more complex and doesn't bring performance benefits 🤔